functions/create_draft/interactivity_handler.ts

```diff
@@ -73,6 +74,40 @@ export const openDraftEditView: BlockActionHandler<
       });
     }
   }
+  // If the user selects to discard the draft message
+  if (action.selected_option.value == "discard_message_overflow") {
+    const id = action.block_id;
+    const thread_ts = body.message?.ts || "";
+
+    // Delete the draft message from the Draft Channel
+    const updateResp = await client.chat.delete({
+      channel: inputs.channel,
+      ts: thread_ts,
+    });
+
+    if (!updateResp.ok) {
+      const updateDraftPreviewErrorMsg =
+        `Error deleting the draft message: ${thread_ts} in channel ${inputs.channel}. Contact the app maintainers with the following - (Error detail: ${updateResp.error})`;
+      console.log(updateDraftPreviewErrorMsg);
+    }
+
+    // Delete the draft from the 'drafts' Datstore
+    const deleteResp = await client.apps.datastore.delete({
+      datastore: DraftDatastore.name,
+      id: id,
+    });
```

Comment:
Without getting too deep in thought, I might opt to delete from the datastore before deleting the message in case something goes wrong with the datastore. In either case an error from either of these gets us into a weird state, so I don't think this has to be changed, but would rather the datastore act as a source of drafting truths if possible!

Reply:
Ahh yeah that makes sense.  I'll move the Delete Datastore operation before the Slack Message Deletion 👌